hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializer.java

```diff
@@ -225,17 +229,20 @@ private CompactSerializableRegistration getOrCreateRegistration(String typeName)
         return typeNameToRegistrationMap.computeIfAbsent(typeName, s -> {
             Class<?> clazz;
             try {
-                //when the registration does not exist, we treat typeName as className to check if there is a class
-                //with the given name in the classpath.
+                // When the registration does not exist, we treat typeName as className
+                // to check if there is a class with the given name in the classpath.
                 clazz = ClassLoaderUtil.loadClass(classLoader, typeName);
             } catch (Exception e) {
-                return null;
-            }
-            try {
-                return getOrCreateRegistration(clazz);
-            } catch (Exception e) {
-                throw new HazelcastSerializationException("Class " + clazz + " must have an empty constructor", e);
```

Comment:
Why did we have this catch block? As far as I see, if ClassLoaderUtil.loadClass does not return null (which I believe cannot) this getOrCreateRegistration call won't throw.

Reply:
This may be a left-over. We used to require empty constructor but later we found an internal way to instantiate the class even if it does not have empty constructor. It can be removed.